### ISSUE TITLE
Inserted items are evicted immediately if no `maxSize` is set

### ISSUE THREAD

**Issue by dkaps125** [isaacs/node-lru-cache#257]
I'm not sure if this is intended, but it's definitely bizarre behavior given that this parameter is optional. See the following example:
```
const cache = LRU({
  max: 500,
  maxEntrySize: 75,
  sizeCalculation: (_value, key) => key.length,
  fetchMethod(key) {
    return `${Math.random()}`
  }
});

console.log(await cache.fetch('test')); // prints a random decimal
console.log(await cache.has('test')); // prints false
```

I think [this while loop](https://github.com/isaacs/node-lru-cache/blob/main/index.js#L404) should only be run when `maxSize` is defined, and `maxSize` shouldn't default to 0.

**Comment by isaacs** [comment]
Yeah, that's a bug. Good catch. Should only skip saving entries bigger than maxEntrySize, not evict unless there's a maxSize set. On Nov 1, 2022, at 18:03, Daniel ***@***.***> wrote:﻿
I'm not sure if this is intended, but it's definitely bizarre behavior given that this parameter is optional. See the following example:
const cache = LRU({
  max: 500,
  maxEntrySize: 75,
  sizeCalculation: (_value, key) => key.length,
  fetchMethod(key) {
    return `${Math.random()}`
  }
});

console.log(await cache.fetch('test')); // prints a random decimal
console.log(await cache.has('test')); // prints false

I think this while loop should only be run when maxSize is defined, and maxSize shouldn't default to 0.

**Comment by dkaps125** [comment]
@isaacs thanks for confirming. I'll put up a PR to address in the next day or so.

**Comment by isaacs** [comment]
Thanks, fixed on 7.14.1 (took your commit, added a test and fixed a weird interaction with this and async fetching, where it was trying to calculate the length of background fetch promises).

**Comment by dkaps125** [comment]
Awesome, thanks for closing this out so quickly!
